### PR TITLE
1st codebase for mentor review

### PR DESCRIPTION
**Pull request by Minh-Tester**
https://dev.azure.com/RookiesBatchVI/RookiesBatch6Group4/_workitems/edit/404

### CODE REVIEW

**Review by honganh1206 — DISMISSED** [review]
LGTM!

File: Repo_group4msauto.sln [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿
```
Comment (daipn1995) [inline Repo_group4msauto.sln]: The file name should be the same with repo name group4msauto.sln

File: Automation_Test_Framework/DAO/AssetListDAO.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Automation_Test_Framework.DAO;
```
Comment (honganh1206) [inline Automation_Test_Framework/DAO/AssetListDAO.cs]: Need DAO for Returning and Report

File: Automation_Test_Framework/PageObject/HomePage.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using Automation_Test_Framework.TestSetup;
+using Core_Framework.DriverCore;
+using OpenQA.Selenium;
+
+namespace Automation_Test_Framework.PageObject;
```
Comment (honganh1206) [inline Automation_Test_Framework/PageObject/HomePage.cs]: Need more Page Objects for Creating + Editing + Logout ...

File: Automation_Test_Framework/TestSetData/TestSetDataAsset.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            if (asset.AssetCode.Contains(" "))
+            {
+                listOfAsset.Remove(asset);
+            }
+            else
+            {
+                continue;
+            }
+        }
+        string assetList = (string)ConvertToJson(listOfAsset);
```
Comment (honganh1206) [inline Automation_Test_Framework/TestSetData/TestSetDataAsset.cs]: This should be a separate function (a function to return DAO object, and another to return DAO converted to JSON string)

File: Automation_Test_Framework/DAO/AssignmentListDAO.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public string? No { get; set; }
+    public string? AssetCode { get; set; }
+    public string? AssetName { get; set; }
+    public string? Assignedto { get; set; }
+    public string? Assignedby { get; set; }
+    public string? AssignedDate { get; set; }
+    public string? State { get; set; }
+
+    public AssignmentListDAO(string assetcode, string assetname, string assignedto, string assignedby,
```
Comment (honganh1206) [inline Automation_Test_Framework/DAO/AssignmentListDAO.cs]: params should be assetCode/assetName/...

File: Automation_Test_Framework/PageObject/LoginPage.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Automation_Test_Framework.PageObject;
+
+public class LoginPage : WebDriverAction
+{
+    public LoginPage(IWebDriver driver) : base(driver)
+    {
+    }
+
+    //private readonly string btnLogin = "";
```
Comment (honganh1206) [inline Automation_Test_Framework/PageObject/LoginPage.cs]: You can log in our project page and check these locators now!

**Review by honganh1206 — APPROVED** [review]
LGTM!

File: group4mauto.sln [hunk 7]
```diff
@@ -0,0 +1,31 @@
+﻿
```
Comment (daipn1995) [inline group4mauto.sln]: Incorrect solution file name

File: README.md [hunk 8]
```diff
@@ -1 +1,155 @@
...
+
+Tests examples
+
+```bash
+*** Settings ***
+Documentation    Verify search functionality
+Resource    ../Pages/SearchResultPage.robot
+Resource    ../Pages/LandingPage.robot
+Resource    ../Helperes/Common.robot
+
```
Comment (daipn1995) [inline README.md]: Incorrect readme doc

File: Core_Framework/Reporter/HtmlReport.cs [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    {
+        GetTest().Info(MarkupHelper.CreateCodeBlock(code, CodeLanguage.Xml));
+    }
+    // ------------------------------- API  -------------------------------
+    public static void CreateAPIRequestLog(APIRequest request, APIResponse response)
+    {
+        GetTest().Info(MarkupHelperPlus.CreateAPIRequestLog(request, response));
+    }
+
+}
```
Comment (daipn1995) [inline Core_Framework/Reporter/HtmlReport.cs]: Add the end line

File: Core_Framework/Reporter/HtmlReport.cs [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+namespace Core_Framework.Reporter;
+
+internal class HtmlReport
+{
+    private static ExtentReports? _report;
+    private static ExtentTest? _extentTestSuite;
+    private static ExtentTest? _extentTestCase;
+
+    // ------------------------------- CREATE EXTENTREPORT  -------------------------------
```
Comment (daipn1995) [inline Core_Framework/Reporter/HtmlReport.cs]: If you want to separate the class part, please use the region instead

image

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    protected ExtentTest? _extentTestCase;
+
+    [OneTimeSetUp]
+    public void OneTimeSetUp()
+    {
+        HtmlReport.CreateReport();
+        HtmlReport.CreateTest(TestContext.CurrentContext.Test.ClassName)
+            .AssignAuthor("Tran Nguyen Minh")
+            .AssignDevice("PC")
+            .AssignCategory("Selenium Project");
```
Comment (daipn1995) [inline Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Don't be hardcoded. Please use the appsettings file or create a class to store these instances

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+        WebDriverManager.InitDriver("chrome", 1920, 1080);
+        Driver = WebDriverManager.GetCurrentDriver();
+        DriverBaseAction = new WebDriverBase(Driver);
+
+    }
+
+    [TearDown]
+    public void TearDown()
+    {
+        Driver?.Quit();
```
Comment (daipn1995) [inline Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: You've already close driver by WebDriverManager.CloseDriver();

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+        TestStatus testStatus = TestContext.CurrentContext.Result.Outcome.Status;
+        if (testStatus.Equals(TestStatus.Passed))
+        {
+            //_extentTestCase?.Pass($"[Passed] Test {TestContext.CurrentContext.Test.Name}");
+        }
+        else if (testStatus.Equals(TestStatus.Failed))
+        {
+            //_extentTestCase?.Fail($"[Failed] Test {TestContext.CurrentContext.Test.Name} because the error \n - {TestContext.CurrentContext.Test.Name}");
+        }
```
Comment (daipn1995) [inline Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Do smt in the if else condition or remove it

File: Core_Framework/DriverCore/WebDriverManager.cs [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    //    else if (Browser.SequenceEqual("safari"))
+    //    {
+    //        Driver = new SafariDriver();
+    //    }
+    //    Driver.Manage().Window.Maximize();
+    //    Driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);
+    //    Driver.Manage().Window.Size = new Size(ScreenWidth, ScreenHeight);
+    //    return Driver;
+    //}
+
```
Comment (daipn1995) [inline Core_Framework/DriverCore/WebDriverManager.cs]: Remove all unused code or meaningless comments for all files changed

File: Core_Framework/API_Core/APIResponse.cs [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    // ------------------------------- METHODS -------------------------------
+
+    //public int GenerateID()
+    //{
+    //    var jsonData = JSON.parse(responseBody);
+    //    System.Reflection.PropertyInfo pi = jsonData.GetType().GetProperty("todos");
+    //    Array name = (Array)(pi.GetValue(0, null));
+    //    int id = name[0].id;
+    //    return id;
```
Comment (daipn1995) [inline Core_Framework/API_Core/APIResponse.cs]: Remove this and other unused code

File: Core_Framework/API_Core/APIResponse.cs [hunk 16]
```diff
@@ -0,0 +1,73 @@
+﻿using System.Net;
+
+namespace Core_Framework.API_Core;
+
+public class APIResponse
+{
+
+    // ------------------------------- ATTRIBUTES -------------------------------
+
+    public HttpWebResponse response;
```
Comment (daipn1995) [inline Core_Framework/API_Core/APIResponse.cs]: Response

File: Automation_Test_Framework/TestSetup/NUnitWebTestSetUp.cs [hunk 17]
```diff
@@ -0,0 +1,19 @@
+﻿using Core_Framework.NUnitTestSetup;
+using NUnit.Framework;
+
+namespace Automation_Test_Framework.TestSetup;
+
+public class NUnitWebTestSetUp : NUnit_Test_Setup
```
Comment (daipn1995) [inline Automation_Test_Framework/TestSetup/NUnitWebTestSetUp.cs]: NUnitWebTestSetUp NUnitAPITestSetUp NUnitWebAndAPITestSetUp 
They have the same purpose. So please don't divide the base test into many parts

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+using Core_Framework.Reporter;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using OpenQA.Selenium;
+
+namespace Core_Framework.NUnitTestSetup;
+
+[TestFixture]
+public class NUnit_Test_Setup
```
Comment (daipn1995) [inline Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Fortmat class name as CamelCase

File: Core_Framework/API_Core/APIRequest.cs [hunk 19]
```diff
@@ -0,0 +1,181 @@
+﻿using Core_Framework.Reporter;
+using System.Net;
+using System.Text;
+
+namespace Core_Framework.API_Core;
```
Comment (daipn1995) [inline Core_Framework/API_Core/APIRequest.cs]: Format all folder format as the CamelCase

File: Automation_Test_Framework/TestSetData/TestSetDataAsset.cs [hunk 20]
```diff
@@ -0,0 +1,61 @@
+﻿using Core_Framework.DriverCore;
+using OpenQA.Selenium;
+using TheRookiesApp.DAO;
+
+namespace TheRookiesApp.TestSetData;
+
+public class TestSetDataAsset : WebDriverBase
+{
+    public TestSetDataAsset(IWebDriver? driver) : base(driver)
```
Comment (daipn1995) [inline Automation_Test_Framework/TestSetData/TestSetDataAsset.cs]: Why do we need to inherit the WebDriverBase and define driver here?

File: Automation_Test_Framework/Common/Common_Flow.cs [hunk 21]
```diff
@@ -0,0 +1,25 @@
+﻿using OpenQA.Selenium;
+using TheRookiesApp.PageObject;
+using TheRookiesApp.TestSetup;
+
+namespace TheRookiesApp.Common
+{
+    public class CommonFlow
```
Comment (daipn1995) [inline Automation_Test_Framework/Common/Common_Flow.cs]: Summary the class purpose as a comment
/// <sumary>
This class is TestSteps object, not CommonFlow.

File: Automation_Test_Framework/Common/Common_Flow.cs [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+    {
+
+
+        public static void LoginFlowAdmin(IWebDriver _driver)
+        {
+            LoginPage loginPage = new LoginPage(_driver);
+            loginPage.DoLogin(Constant.ADMIN_USER_NAME, Constant.ADMIN_PASSWORD);
+        }
```
Comment (daipn1995) [inline Automation_Test_Framework/Common/Common_Flow.cs]: 1. You have to define the page many times. This is not a good coding convention for the base.
2. Just 1 line to log in as admin. So why do we need to create this method

File: Automation_Test_Framework/Common/Common_Flow.cs [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+
+        public static void LoginFlowStaff(IWebDriver _driver)
+        {
+            LoginPage loginPage = new LoginPage(_driver);
+            loginPage.DoLogin(Constant.STAFF_USER_NAME, Constant.STAFF_PASSWORD);
+        }
```
Comment (daipn1995) [inline Automation_Test_Framework/Common/Common_Flow.cs]: Same as above

File: Automation_Test_Framework/Asserter/Asserter.cs [hunk 24]
```diff
@@ -0,0 +1,168 @@
...
+
+namespace AssetManagementTestProject.Asserter;
+/// <summary>
+/// Assertions for strings/DAO/List DAO objects
+/// Also check order
+/// Alternatives if FluentAssertions fails
+/// </summary>
+public class Asserter : WebDriverBase
+{
+    public Asserter(IWebDriver driver) : base(driver)
```
Comment (daipn1995) [inline Automation_Test_Framework/Asserter/Asserter.cs]: ??? why asserter is inherited by WebDriverBase

File: Automation_Test_Framework/Asserter/Asserter.cs [hunk 25]
```diff
@@ -0,0 +1,168 @@
...
+        // Compare two deserialized JSON strings
+        AssertEquals(actual, expected);
+        //actual.Should().BeEquivalentTo(expected);
+    }
+    // ------------------------------- DAO LIST WITH FLUENT ASSERTION -------------------------------
+
+    public static void AssertUserListsEquals(List<UserListDAO> actual, List<UserListDAO> expected)
+    {
+        AssertEquals(actual, expected);
+        //actual.Should().BeEquivalentTo(expected);
```
Comment (daipn1995) [inline Automation_Test_Framework/Asserter/Asserter.cs]: Is there any meaning to use the fluent assertion here?
You can use the hook instead to log HTML report. Don't do this way to just log the html log

File: Automation_Test_Framework/Asserter/AssetListAsserter.cs [hunk 26]
```diff
@@ -0,0 +1,39 @@
+﻿//using Core_Framework.DriverCore;
+//using NUnit.Framework;
+//using OpenQA.Selenium;
+//using TheRookiesApp.DAO;
+
+//// ------------------------------- IN CASE FLUENT ASSERTION FAILS -------------------------------
```
Comment (daipn1995) [inline Automation_Test_Framework/Asserter/AssetListAsserter.cs]: Remove all unused classes or code

File: .gitignore [hunk 27]
```diff
@@ -0,0 +1,407 @@
+# Created by https://www.toptal.com/developers/gitignore/api/visualstudio
```
Comment (daipn1995) [inline .gitignore:1]: ignore reporters folder

File: group4mauto.sln (hunk 7 above)
Comment (Minh-Tester) [reply group4mauto.sln]: Done!

File: Core_Framework/Reporter/HtmlReport.cs (hunk 10 above)
Comment (Minh-Tester) [reply Core_Framework/Reporter/HtmlReport.cs]: Done!

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs (hunk 12 above)
Comment (Minh-Tester) [reply Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Done!

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs (hunk 13 above)
Comment (Minh-Tester) [reply Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Done!

File: Core_Framework/DriverCore/WebDriverManager.cs (hunk 14 above)
Comment (Minh-Tester) [reply Core_Framework/DriverCore/WebDriverManager.cs]: Done!

File: Core_Framework/API_Core/APIResponse.cs (hunk 15 above)
Comment (Minh-Tester) [reply Core_Framework/API_Core/APIResponse.cs]: Done!

File: Core_Framework/API_Core/APIResponse.cs (hunk 16 above)
Comment (Minh-Tester) [reply Core_Framework/API_Core/APIResponse.cs]: Done!

File: Automation_Test_Framework/TestSetup/NUnitWebTestSetUp.cs (hunk 17 above)
Comment (Minh-Tester) [reply Automation_Test_Framework/TestSetup/NUnitWebTestSetUp.cs]: Done!

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs (hunk 18 above)
Comment (Minh-Tester) [reply Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Done!

File: Core_Framework/API_Core/APIRequest.cs (hunk 19 above)
Comment (Minh-Tester) [reply Core_Framework/API_Core/APIRequest.cs]: Done!

File: Automation_Test_Framework/TestSetData/TestSetDataAsset.cs (hunk 20 above)
Comment (Minh-Tester) [reply Automation_Test_Framework/TestSetData/TestSetDataAsset.cs]: Done!

File: Automation_Test_Framework/Common/Common_Flow.cs (hunk 21 above)
Comment (Minh-Tester) [reply Automation_Test_Framework/Common/Common_Flow.cs]: Done!

File: Automation_Test_Framework/Asserter/Asserter.cs (hunk 24 above)
Comment (Minh-Tester) [reply Automation_Test_Framework/Asserter/Asserter.cs]: I define AssertEquals in WebDriverBase so Asserter inherit WebDriverBase

File: Core_Framework/DriverCore/WebDriverBase.cs [hunk 28]
```diff
@@ -0,0 +1,569 @@
...
+            HtmlReport.Pass("Actual data [" + actual.ToString() + "] matches " +
+                "with expected data [" + expected.ToString() + "]");
+        }
+        catch (Exception excep)
+        {
+            HtmlReport.Fail("Actual data [" + actual + "] does not match " +
+                "with expected data [" + expected + "]");
+            throw excep;
+        }
+    }
```
Comment (daipn1995) [inline Core_Framework/DriverCore/WebDriverBase.cs]: This method does not belong to WebDriverbase. Please move it to another object like CustomAssertion or smt like that

File: group4msauto.sln [hunk 29]
```diff
@@ -0,0 +1,31 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.3.32929.385
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TheRookiesApp", "Automation_Test_Framework\TheRookiesApp.csproj", "{ADDC31AA-4713-4926-901C-F08C0A4840B9}"
```
Comment (daipn1995) [inline group4msauto.sln]: The folder name should be CamelCase and the project name should be the same with the folder name

File: group4msauto.sln [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Core_Framework", "Core_Framework\Core_Framework.csproj", "{7C2D75CE-B288-4910-8DBD-6C6D6F457988}"
```
Comment (daipn1995) [inline group4msauto.sln]: Same as above

File: Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs (hunk 18 above)
Comment (daipn1995) [reply Core_Framework/NUnitTestSetup/NUnit_Test_Setup.cs]: Not done yet

File: group4msauto.sln (hunk 29 above)
Comment (Minh-Tester) [reply group4msauto.sln]: Does this error affect the codebase?

Comment (Minh-Tester) [reply group4msauto.sln]: I fixed it. Done

File: group4msauto.sln (hunk 30 above)
Comment (Minh-Tester) [reply group4msauto.sln]: Done!

File: TheRookiesApp/Common/TestSteps.cs [hunk 31]
```diff
@@ -0,0 +1,10 @@
+﻿using TheRookiesApp.PageObject;
+using TheRookiesApp.TestSetup;
+
+namespace TheRookiesApp.Common
+{
+    public class TestSteps
+    {
+
+    }
+}
```
Comment (daipn1995) [inline TheRookiesApp/Common/TestSteps.cs]: Add the end line

File: TheRookiesApp/Common/TestSteps.cs [hunk 32]
```diff
@@ -0,0 +1,10 @@
+﻿using TheRookiesApp.PageObject;
+using TheRookiesApp.TestSetup;
+
+namespace TheRookiesApp.Common
+{
+    public class TestSteps
```
Comment (daipn1995) [inline TheRookiesApp/Common/TestSteps.cs]: Add comment for the class purpose

File: Automation_Test_Framework/Asserter/Asserter.cs (hunk 25 above)
Comment (daipn1995) [reply Automation_Test_Framework/Asserter/Asserter.cs]: Not done yet. Remove comment unused commands.
Check other classes

File: TheRookiesApp/Asserter/Asserter.cs [hunk 33]
```diff
@@ -0,0 +1,168 @@
...
+            }
+            catch (Exception e)
+            {
+                Assert.True(false, $"Assertion failed because: {e.Message}");
+            }
+        }
+    }
+    #endregion
+
+}
```
Comment (daipn1995) [inline TheRookiesApp/Asserter/Asserter.cs:178]: Add the end line

Comment (Minh-Tester) [reply TheRookiesApp/Asserter/Asserter.cs:178]: Done

File: TheRookiesApp/Common/TestSteps.cs (hunk 31 above)
Comment (Minh-Tester) [reply TheRookiesApp/Common/TestSteps.cs]: Done

File: TheRookiesApp/Common/TestSteps.cs (hunk 32 above)
Comment (Minh-Tester) [reply TheRookiesApp/Common/TestSteps.cs]: What should I do if I don't know what is the purpose of this class?

**Comment by Minh-Tester** [comment]
image

File: TheRookiesApp/Common/TestSteps.cs (hunk 32 above)
Comment (daipn1995) [reply TheRookiesApp/Common/TestSteps.cs]: Then remove it

File: README.md [hunk 34]
```diff
@@ -1 +1,162 @@
...
+namespace TheRookiesApp.TestCase;
+
+[TestFixture]
+public class US302LoginTest : NUnitTestSetUp
+{
+    [Test]
+    public void ID01LoginAndLogoutTest()
+    {
+        TestSteps.LoginAsAdmin();
```
Comment (daipn1995) [inline README.md]: Where is the TestSteps?

Comment (daipn1995) [reply README.md]: Incorrect guideline

File: README.md [hunk 35]
```diff
@@ -1 +1,162 @@
...
+// Variables 
+public IWebDriver Driver;
+public IJavaScriptExecutor Javascript { get; set; }
+private WebDriverWait _explicitWait;
+private Actions _actions;
+private int _timeWait = 60;
+
+// Methods
+public void MoveForward()
+{
```
Comment (daipn1995) [inline README.md]: This is not the page sample. You should write by yourself. Don't try to copy 100% of the idea if you don't understand what you're writing for

File: TheRookiesApp/PageObject/MenuLeft.cs [hunk 36]
```diff
@@ -0,0 +1,25 @@
...
+    //{
+    //    if (Constant.BASE_URL + addedURL == GetUrl())
+    //    {
+    //        return true;
+    //    }
+    //    else
+    //    {
+    //        return false;
+    //    }
+    //}
```
Comment (daipn1995) [inline TheRookiesApp/PageObject/MenuLeft.cs]: Remove all unused code. Please check again for the other files

File: TheRookiesApp/Service/UserLogin.cs [hunk 37]
```diff
@@ -0,0 +1,9 @@
+﻿namespace TheRookiesApp.Service;
+
+public class UserLogin
+{
+    public int Id { get; set; }
+    public string? Username { get; set; }
+    public string? Password { get; set; }
+    public string? Token { get; set; }
+}
```
Comment (daipn1995) [inline TheRookiesApp/Service/UserLogin.cs]: Again. Add the end line

File: CoreFramework/NUnitTestSetup/NUnitSetup.cs [hunk 38]
```diff
@@ -0,0 +1,63 @@
...
+    public void TearDown()
+    {
+        WebDriverManager.CloseDriver();
+
+        TestStatus testStatus = TestContext.CurrentContext.Result.Outcome.Status;
+        if (testStatus.Equals(TestStatus.Passed))
+        {
+            HtmlReport.Pass("PASSED: Test case passed");
+        }
+        else if (testStatus.Equals(TestStatus.Failed))
```
Comment (daipn1995) [inline CoreFramework/NUnitTestSetup/NUnitSetup.cs]: else not else if

File: CoreFramework/NUnitTestSetup/NUnitSetup.cs [hunk 39]
```diff
@@ -0,0 +1,63 @@
...
+        {
+                HtmlReport.Fail("FAILED: Test errors: "+ TestContext.CurrentContext.Result.Message, DriverBaseAction.TakeScreenShot());
+        }
+
```
Comment (daipn1995) [inline CoreFramework/NUnitTestSetup/NUnitSetup.cs]: remove empty line

File: CoreFramework/Reporter/ExtentMarkup/API_Request_Log.cs [hunk 40]
```diff
@@ -0,0 +1,32 @@
...
+                <p>Response status: {Response.ResponseStatusCode} <p>
+            ";
+        return log;
+
+    }
+
+    /*public static void CreateAPIRequestLog(API_Request request, API_Response response)
+    {
+        GetTest().Info(Markup_Helper_Plus.CreateAPIRequestLog(request, response));
+    }*/
```
Comment (daipn1995) [inline CoreFramework/Reporter/ExtentMarkup/API_Request_Log.cs:31]: ?

File: TheRookiesApp/Common/TestSteps.cs (hunk 32 above)
Comment (Minh-Tester) [reply TheRookiesApp/Common/TestSteps.cs]: Done!

File: README.md (hunk 34 above)
Comment (Minh-Tester) [reply README.md]: I updated it in the latest codebase

File: TheRookiesApp/Service/UserLogin.cs (hunk 37 above)
Comment (Minh-Tester) [reply TheRookiesApp/Service/UserLogin.cs]: Done

File: CoreFramework/NUnitTestSetup/NUnitSetup.cs (hunk 39 above)
Comment (Minh-Tester) [reply CoreFramework/NUnitTestSetup/NUnitSetup.cs]: Done!

File: CoreFramework/Reporter/ExtentMarkup/API_Request_Log.cs (hunk 40 above)
Comment (Minh-Tester) [reply CoreFramework/Reporter/ExtentMarkup/API_Request_Log.cs:31]: Done

File: CoreFramework/NUnitTestSetup/NUnitSetup.cs (hunk 38 above)
Comment (Minh-Tester) [reply CoreFramework/NUnitTestSetup/NUnitSetup.cs]: Done

File: TheRookiesApp/PageObject/MenuLeft.cs (hunk 36 above)
Comment (Minh-Tester) [reply TheRookiesApp/PageObject/MenuLeft.cs]: Done

**Comment by daipn1995** [comment]
PR still has some errors. Close this PR!

**Pull request closed without merging**